### PR TITLE
implement macros & includes

### PR DESCRIPTION
**Pull request by seletskiy**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,10 +30,127 @@ File in extended format should follow specification
...
+be replaced with specified template:
+
+```markdown
+<!-- Macro: <regexp>
+     Template: <path>
+     <yaml-data> -->
+```
+
+Capture groups can be defined in the macro's `<regexp>` which can be later
+referenced in the `<yaml-data>` using `${<number>}` syntax.
```
Comment (kovetskiy) [inline README.md]: Didn't get, what the `<number>` here? It's mentioned here first time

File: main.go [hunk 2]
```diff
@@ -117,50 +137,85 @@ func main() {
...
-	if dryRun {
-		fmt.Println(string(mark.CompileMarkdown(markdownData)))
-		os.Exit(0)
+	meta, err := mark.ExtractMeta(markdown)
+	if err != nil {
+		log.Fatal(err)
 	}
 
-	creds, err := GetCredentials(args, config)
+	stdlib, err := stdlib.New(api)
```
Comment (kovetskiy) [inline main.go]: can we do something less abstract? maybe longer name but more obvious about purposes of the package

File: pkg/mark/includes/templates.go [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+	"text/template"
+
+	"gopkg.in/yaml.v2"
+
+	"github.com/kovetskiy/mark/pkg/log"
+	"github.com/reconquest/karma-go"
+)
+
+var (
+	reIncludeDirective = regexp.MustCompile(`(?s)<!-- Include: (\S+)(.*?)-->`)
```
Comment (kovetskiy) [inline pkg/mark/includes/templates.go]: The first space can be omitted too in classic HTML
`<!--\s*Include:`

File: pkg/mark/macro/macro.go [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+	"text/template"
+
+	"github.com/kovetskiy/mark/pkg/log"
+	"github.com/kovetskiy/mark/pkg/mark/includes"
+	"github.com/reconquest/karma-go"
+	"gopkg.in/yaml.v2"
+)
+
+var reMacroDirective = regexp.MustCompile(
+	`(?s)<!-- Macro: ([^\n]+)\n\s*Template: (\S+)\n(.*?)-->`,
```
Comment (kovetskiy) [inline pkg/mark/macro/macro.go]: the first space can be omitted here too

File: pkg/mark/macro/macro.go [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+		contents,
+		func(spec []byte) []byte {
+			if err != nil {
+				return spec
+			}
+
+			groups := reMacroDirective.FindSubmatch(spec)
+
+			var (
+				expr, path, config = groups[1], string(groups[2]), groups[3]
```
Comment (kovetskiy) [inline pkg/mark/macro/macro.go]: please use regexputil-go to avoid this ambiguous indexing

File: pkg/mark/includes/templates.go [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+)
+
+func LoadTemplate(
+	path string,
+	templates *template.Template,
+) (string, *template.Template, error) {
```
Comment (kovetskiy) [inline pkg/mark/includes/templates.go]: that returned `string` which is name is unused in the code

File: pkg/mark/stdlib/stdlib.go [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+			/**/ `<ac:layout>`,
+			/**/ `<ac:layout-section ac:type="two_right_sidebar">`,
+			/**/ `<ac:layout-cell>{{ .Body }}</ac:layout-cell>`,
+			/**/ `<ac:layout-cell></ac:layout-cell>`,
+			/**/ `</ac:layout-section>`,
+			/**/ `</ac:layout>`,
+			`{{ else }}`,
+			/**/ `{{ .Body }}`,
+			`{{ end }}`,
+		),
```
Comment (kovetskiy) [inline pkg/mark/stdlib/stdlib.go]: just a quick note: we could also store these templates in files and distribute them along with binary

File: main.go (hunk 2 above)
Comment (seletskiy) [reply main.go]: I can't come up with a better name. Do you have some variants in mind?

File: pkg/mark/stdlib/stdlib.go (hunk 7 above)
Comment (seletskiy) [reply pkg/mark/stdlib/stdlib.go]: Yep, but on the other hand it will make tool not ready to use after `go get`/`go install`.

**Pull request merged**